File: 3222-find-the-winning-player-in-coin-game/3222-find-the-winning-player-in-coin-game.py
class Solution:
    def losingPlayer(self, x: int, y: int) -> str:
        
        turn = True 

        cnt = 0 

        while x >= 1 and y >= 4:
            x -= 1
            y -= 4
            cnt +=1
        
        if cnt % 2 == 0:
            return "Bob"
        
        return "Alice"

File: 3222-find-the-winning-player-in-coin-game/3222-find-the-winning-player-in-coin-game.js
/**
 * @param {number} x
 * @param {number} y
 * @return {string}
 */
var losingPlayer = function(x, y) {
    return Math.min(Math.floor(x), Math.floor(y / 4)) % 2 ? "Alice" : "Bob";
};